src/main/java/tech/toparvion/analog/service/choice/ChoicesPropertiesChangesListener.java

```diff
@@ -0,0 +1,74 @@
+package tech.toparvion.analog.service.choice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Service;
+import tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.FileWatcherProvider;
+
+import javax.annotation.Nullable;
+import java.nio.file.*;
+
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+import static tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.CHOICES_HOT_RELOAD_EXECUTOR;
+
+/**
+ * @author Polyudov
+ * @since 0.14
+ */
+@Service
+@EnableAsync
+@ConditionalOnChoicesAutoReloadEnabled
+class ChoicesPropertiesChangesListener {
+  private static final Logger log = LoggerFactory.getLogger(ChoicesPropertiesChangesListener.class);
+
+  private final RefreshEndpoint refreshEndpoint;
+  private final FileWatcherProvider fileWatcherProvider;
+
+  @Autowired
+  ChoicesPropertiesChangesListener(RefreshEndpoint refreshEndpoint,
+                                   @Nullable FileWatcherProvider fileWatcherProvider) {
+    this.refreshEndpoint = refreshEndpoint;
+    this.fileWatcherProvider = fileWatcherProvider;
+  }
+
+  @Async(CHOICES_HOT_RELOAD_EXECUTOR)
+  @EventListener(ApplicationReadyEvent.class)
+  public void watchFile() {
+    if (fileWatcherProvider == null) {
+      return;
+    }
+    try {
+      log.info("Start watching for the choices properties file");
+      WatchService watchService = fileWatcherProvider.getWatchService();
+      WatchKey key;
+      while ((key = watchService.take()) != null) {
+        for (WatchEvent<?> event : key.pollEvents()) {
+          //Filter some special events and repeated events
+          if (OVERFLOW.equals(event.kind()) || event.count() > 1) {
+            continue;
+          }
+          Path path = (Path) event.context();
+          //Check modified file
+          if (fileWatcherProvider.isChoicesPropertiesFileEvent(path)) {
+            refreshEndpoint.refresh();
```

Comment:
Do you think it will be some usefull information?
Example:
```
15:17:28.969  INFO     hot-reload1 - t.a.s.c.ChoicesPropertiesChangesListener : Choices properties file was reloaded
15:17:28.969 TRACE     hot-reload1 - t.a.s.c.ChoicesPropertiesChangesListener : Refreshed properties keys: [choices.list[0].compositeLogs[0].title, choices.list[0].compositeLogs[0].includes[1].timestamp, choices.list[0].group, choices.list[0].compositeLogs[0].includes[0].path, choices.list[0].plainLogs[3].path, choices.list[0].plainLogs[0].title, choices.list[0].compositeLogs[0].includes[1].path, choices.list[0].plainLogs[0].path, choices.list[0].plainLogs[3].selected, choices.list[0].plainLogs[2].selected, choices.list[0].plainLogs[1].path, choices.list[0].localPlainLogsBase, choices.list[0].compositeLogs[0].uriName, choices.list[0].plainLogs[2].path, choices.list[0].compositeLogs[0].includes[0].timestamp]
```

Reply:
Umph, that's really not of much help. Ok, let's keep omitting this result.